Src/BootCamp.Chapter/Lesson4.cs

```diff
@@ -0,0 +1,126 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace BootCamp.Chapter
+{
+    class Lesson4
+    {
+        public static void Demo()
+        {
+            //main
+            Interview();
+        }
+
+        public static void Interview()
+        {
+            //  gets all the input from the user, if one of them is invalid
+            //  it displays the relevant error code and returns
+            string name = PromptString("Name: ");
+            if (ErrorString(name, "Name cannot be empty.")) return;
+            
+            string surname = PromptString("Surname: ");
+            if (ErrorString(surname, "Surname cannot be empty.")) return;
+
+            int age = PromptInt("Age: ");
+            if (age < 0) return;
+
+            float weight = PromptFloat("Weight(Kg): ");
+            if (weight < 0) return;
+
+            float cmHeight = PromptFloat("Height (cm): ");
+            if (cmHeight < 0) return;
+            float mHeight = cmHeight / 100f;
+
+            //  prints the description according to user's input
+            Console.WriteLine($"{name} {surname} is {age} years old, their weight is {weight}Kg and their height is {cmHeight}cm");
+
+            //  calculate BMI and print errormessage if it got an error
+            float bmi = CalculateBmi(weight, mHeight);
+            if (bmi < 0)
+            {
+                Console.WriteLine("Falied to calculate BMI. Reasons:");
+                if ((bmi == -2)||(bmi == -3))
+                {
+                    Console.WriteLine($"Height cannot be equal or less than zero, but was {cmHeight}.");
+                }
+                if ((bmi == -1) || (bmi == -3))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Do not check floating point equality with exact values, use a range instead.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4876897)